src/admin/class.admin.php

```diff
@@ -0,0 +1,219 @@
+            // Validate file type and size
+            if (
+                $file['type'] !== 'text/csv' ||
+                $file['size'] > 2097152 // 2MB limit
+            ) {
+                throw new Exception("Invalid file type or size.");
+            }
+    
+            // Ensure target directory exists
+            $uploadsDir = 'uploads';
+            if (!is_dir($uploadsDir) && !mkdir($uploadsDir, 0755, true)) {
+                throw new Exception("Failed to create uploads directory.");
+            }
+    
+            // Move the file
+            if (!move_uploaded_file($file['tmp_name'], "{$uploadsDir}/{$file['name']}")) {
+                throw new Exception("Failed to move uploaded file.");
+            }
+        } catch (Exception $e) {
+            throw new RuntimeException("File handling error: " . $e->getMessage());
+        }
+    }
+
+    /**
+     * @param string $fileName
+     * 
+     * @return array
+     */
+    private function parseCsvData(string $fileName): array
+    {
+        try {
+            $data = array_map('str_getcsv', file('./uploads/' . $fileName));
+            array_shift($data);
+            return $data;
+        } catch (Exception $e) {
+            return [];
+        }
+    }
+
+    /**
+     * @param array $data
+     * 
+     * @return array
+     */
+    private function processRows(array $data): array
+    {
+        $newData = [];
+        foreach ($data as $row) {
+            try {
+                $newRow = [];
+
+                if (!empty($row[6]) && !empty($row[15])) {
+                    foreach (['small', 'medium'] as $size) {
+                        $this->resizeImage($row[6], $row[15], $size);    
+                    }
+                } else {
+                    continue;
+                }                    
+
+                $newRow['name'] = htmlspecialchars("$row[1] $row[3]");
+                $newRow['imageSmall'] = "./entries/images/$row[15]-small.jpg";
+                $newRow['imageMedium'] = "./entries/images/$row[15]-medium.jpg";
+                $newRow['message'] = htmlspecialchars($row[7]);
+                $newRow['highlight'] = false;
+                $newRow['id'] = $row[15];
+        
+                array_push($newData, $newRow);
+
+            } catch (Exception $e) {
+                print_r($e);
+            }
+        }
+    
+        return $newData;
+    }   
+    
+    /**
+     * @param array $data
+     * 
+     * @return void
+     */
+    private function saveDataToJson(array $data): void
+    {
+        try {
+            $json = json_encode($data, JSON_THROW_ON_ERROR);
+            file_put_contents('../entries/data.json', $json);
+        } catch (Exception $e) {
+            print_r($e);
+        }
+    }
+
+    /**
+     * @param string $imageUrl
+     * @param int $entryId
+     * @param string $size
+     * 
+     * @return void
+     */
+    private function resizeImage(string $imageUrl, int $entryId, string $size): void
```

Comment:
Will we or does this center crop the images? I noticed the sample small images isn't center cropped.